app/src/main/res/layout/item_occurrence.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical"
+    android:layout_margin="6dp">
+
+    <TextView
+        android:id="@+id/what"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textColor="?android:textColorPrimary"
+        android:textSize="?attr/status_text_medium"
+        android:textStyle="bold"
+        android:layout_marginBottom="4dp"
+        tools:text="Example crash" />
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal">
+
+        <TextView
+            android:id="@+id/code"
+            android:layout_width="wrap_content"
+            android:minWidth="50dp"
+            android:layout_height="match_parent"
+            android:lines="1"
+            android:textColor="?android:textColorPrimary"
+            android:textSize="?attr/status_text_large"
+            android:paddingEnd="6dp"
+            tools:text="500" />
+
+        <TextView
+            android:id="@+id/whenDate"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textColor="?android:textColorPrimary"
+            android:textSize="?attr/status_text_medium"
+            android:paddingEnd="6dp"
+            tools:text="2048:04:01" />
+
+        <TextView
+            android:id="@+id/duration"
+            android:layout_width="0dp"
+            android:layout_weight="1"
+            android:layout_height="wrap_content"
+            android:textColor="?android:textColorPrimary"
+            android:textSize="?attr/status_text_medium"
+            tools:text="203ms" />
+
+        <TextView
+            android:id="@+id/who"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textColor="?android:textColorPrimary"
+            android:textSize="?attr/status_text_medium"
+            tools:text="Someone" />
+    </LinearLayout>
+
+    <TextView
+        android:id="@+id/trace"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textColor="?android:textColorPrimary"
+        android:textSize="?attr/status_text_medium"
+        android:layout_marginTop="4dp"
+        android:visibility="gone"
+        tools:text="Trace line 1" />
+</LinearLayout>
```

Comment:
I'd always use ConstraintLayout for these kind of layout.
It goes something like this:
- Just need to center or frame something -> FrameLayout
- Just need some items below/next to each other -> LinearLayout
- Everything else -> ContaintLayout because it avoids nesting of other layouts and should be faster

Reply:
Ok, I will try.

(My reasoning would go something like this: a "line" could be considered a group and thus can be grouped. If for example the top level is a vertical linear layout I would know that its contents are "line-ish" things.)

Reply:
Using ConstraintLayout now.

Cursed two more hours and still have no idea how ConstraintLayout works (with respect to "layout_weight"). By chance I found that width="0dp" does it...